### ISSUE TITLE
[Bug]: Data source hcloud_network does not properly resolve dependencies

### ISSUE THREAD

**Issue by FriesPascal** [hetznercloud/terraform-provider-hcloud#513]
### What happened?

When using the data source `hcloud_network` to read infos about a network that is created in the same run as the data source, `ip_range`, `labels`, `name`, and `delete_protection` are returned immediately as `null` instead of reading them from the network. I think that is because they are not listed as `computed` in https://github.com/hetznercloud/terraform-provider-hcloud/blob/master/internal/network/data_source.go#L27.

This becomes a problem, when trying to define a module, that lets you pass in network ids, and attaches servers to these networks and adds allow-rules to a corresponding firewall.

### What did you expect to happen?

`data.hcloud_network` should mark `ip_range`, `labels`, `name`, and `delete_protection` as `(known after apply)` when the network is read via id and does not exist until the next apply.

### Please provide a minimal working example

```
resource "hcloud_network" "this" {
  name = "test"
  ip_range = "10.0.0.0/24"
}

data "hcloud_network" "this" {
  id = hcloud_network.this.id
}

# contains several instances of null :(
output "debug" {
  value = data.hcloud_network.this
}
```

**Comment by apricote** [comment]
Thanks for the report and sorry for letting it go stale. I can confirm the issue and pushed a fix in #805.

**Comment by jooola** [comment]
Closing as solved by #805
